### PR TITLE
DOCS-530: Refactor Glossary

### PR DESCRIPTION
**Pull request by npentrel**
# Description

Please include a summary of the change and what is does. Please also include relevant motivation and context.

### CODE REVIEW

File: docs/services/frame-system.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ In this page, we will explain:
 
 ## Configuration
 
-To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the Frame group where you will enter Reference Frame details.
-This opens the Frame group:
+To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the gRoup where you will enter Reference Frame details.
```
Comment (npentrel) [inline docs/services/frame-system.md:22]:
```suggestion
To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the Framge group where you will enter Reference Frame details.
```

File: docs/services/frame-system.md [hunk 2]
```diff
@@ -19,8 +19,8 @@ In this page, we will explain:
...
+This opens the gRoup:
```
Comment (npentrel) [inline docs/services/frame-system.md:23]:
```suggestion
This opens the Frame group:
```

File: docs/services/frame-system.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ In this page, we will explain:
 
 ## Configuration
 
-To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the Frame group where you will enter Reference Frame details.
+To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the Framge group where you will enter Reference Frame details.
```
Comment (npentrel) [inline docs/services/frame-system.md:22]:
```suggestion
To supply reference frame information when configuring a component in the Viam app ([https://app.viam.com](https://app.viam.com)), click **Add Frame** to launch the Frame group where you will enter Reference Frame details.
```

File: docs/appendix/glossary/sdk.md [hunk 4]
```diff
@@ -2,16 +2,12 @@
...
-- One per language.
-- Can be used as a server for a custom component implementation.
-  - Hosts a gRPC server implementing the Viam Robot API.
-  - That serves functionality for all registered resources.
-- Can be used as a client.
-  - To connect to a robot implementing the Viam Robot API.
-- Effectively, non-golang versions of {{< glossary_tooltip term_id="rdk" text="RDK" >}}’s resource authoring and activation functionality.
+The SDKs wrap the `viam-server` {{< glossary_tooltip term_id="grpc" text="gRPC" >}} {{< glossary_tooltip term_id="viam-robot-api" text="Viam Robot API" >}} and streamline connection, authentication, and encryption.
+
+For more information, see [Program your Robot with Viam's SDKs](../../program/SDK-as-client.md).
```
Comment (npentrel) [inline docs/appendix/glossary/sdk.md:13]:
```suggestion
For more information, see [Program your Robot with Viam's SDKs](../../program/sdk-as-client.md).
```

File: docs/appendix/glossary/board.md [hunk 5]
```diff
@@ -2,10 +2,13 @@
...
+short_description: A board is the signal wire hub of a robot that provides access to GPIO pins, SPI or I2C.
 aka:
 ---
 
-An IO board that is, or is connected to, a robot part and used to express low-level electronics functionality such as GPIO, SPI, or I2C.
+A board is the signal wire hub of a robot that provides access to GPIO pins, SPI or I2C.
 
-Examples of boards include Jetson, Raspberry Pi, Numato, or Arduino. Boards like Jetson and Pi run viam-server and can expose the board {{< glossary_tooltip term_id="component" text="component" >}} itself.
+Examples of boards include Jetson, Raspberry Pi, Numato, or Arduino.
+Boards like Jetson and Pi run viam-server and can expose the board {{< glossary_tooltip term_id="component" text="component" >}} itself.
```
Comment (JessamyT) [inline docs/appendix/glossary/board.md:12]:
```suggestion
Boards like Jetson and Pi run `viam-server` and can expose the board {{< glossary_tooltip term_id="component" text="component" >}} itself.
```
What does this sentence even mean though?

File: docs/appendix/glossary/frame.md [hunk 6]
```diff
@@ -2,10 +2,10 @@
 title: Frame
 id: frame
 full_link:
-short_description: A single element of a Frame System.
+short_description: A frame represents a coordinate system that describes position and orientation of an object in relation to the Frame System.
 aka:
 ---
 
-A single element of a Frame System.
+A frame represents a coordinate system that describes position and orientation of an object in relation to the {{< glossary_tooltip term_id="frame-system" text="Frame System" >}}.
```
Comment (JessamyT) [inline docs/appendix/glossary/frame.md:9]:
```suggestion
A frame represents a coordinate system that describes position and orientation of an object.
See also {{< glossary_tooltip term_id="frame-system" text="Frame System" >}}.
```
See comment above

File: docs/appendix/glossary/gantry.md [hunk 7]
```diff
@@ -2,8 +2,8 @@
 title: Gantry
 id: attribute
 full_link: /components/gantry/
-short_description: A robot that only uses linear motion to carry out a task; for example, the scaffolding of a 3D printer, which moves the print head around on motorized linear rails.
+short_description: A mechanical system that only uses linear motion to carry out a task.
 aka:
 ---
 
-A robot that only uses linear motion to carry out a task; for example, the scaffolding of a 3D printer, which moves the print head around on motorized linear rails.
+A mechanical system that only uses linear motion to carry out a task; for example, the scaffolding of a 3D printer, which moves the print head around on motorized linear rails.
```
Comment (JessamyT) [inline docs/appendix/glossary/gantry.md:9]: ahhhh *motorized* linear rails...a different solution to the old gantry clarification stuff :)

File: docs/appendix/glossary/model.md [hunk 8]
```diff
@@ -2,8 +2,9 @@
 title: Model
 id: model
 full_link:
-short_description: A particular implementation of a component type. For example, UR5e is a model of the arm component type.
+short_description: A particular implementation of a component type.For example, UR5e is a model of the arm component type.
```
Comment (JessamyT) [inline docs/appendix/glossary/model.md:5]:
```suggestion
short_description: A particular implementation of a component type. For example, `UR5e` is a model of the arm component type.
```
Space is important. Backticks are opt, and do they even render in descriptions?

File: docs/appendix/glossary/rdk.md [hunk 9]
```diff
@@ -1,9 +1,11 @@
 ---
 title: RDK (Robot Development Kit)
 id: rdk
-full_link:
-short_description: The official Viam-developed codebase that provides all functionality of an SDK and more.
+full_link: /program/rdk/
+short_description: The open-source, on-robot portion of the Viam platform, that provides viam-server and the Go SDK.
```
Comment (JessamyT) [inline docs/appendix/glossary/rdk.md:5]:
```suggestion
short_description: The open-source, on-robot portion of the Viam platform, that provides `viam-server` and the Go SDK.
```

File: docs/appendix/glossary/rdk.md [hunk 10]
```diff
@@ -1,9 +1,11 @@
...
 aka:
 ---
 
-The official Viam-developed codebase that provides all functionality of an SDK and [more](/program/rdk/).
+Viam’s RDK is the open-source, on-robot portion of the Viam platform, that provides `viam-server` and the Go SDK.
```
Comment (JessamyT) [inline docs/appendix/glossary/rdk.md:9]:
```suggestion
Viam’s Robot Development Kit (RDK) is the open-source, on-robot portion of the Viam platform, that provides `viam-server` and the Go SDK.
```
[opt]

File: docs/appendix/glossary/resource-config.md [hunk 11]
```diff
@@ -2,8 +2,10 @@
 title: Resource Config
 id: resource-config
 full_link:
-short_description: The configuration element of either a component or a service. Typically expressed in JSON.
+short_description: The configuration element of either a component or a service.
 aka:
 ---
 
-The configuration element of either a {{< glossary_tooltip term_id="component" text="component" >}} or a service. Typically expressed in JSON.
+The configuration element of either a {{< glossary_tooltip term_id="component" text="component" >}} or a {{< glossary_tooltip term_id="service" text="service" >}}.
```
Comment (JessamyT) [inline docs/appendix/glossary/resource-config.md:9]: This glossary item is strange. I would think this term could also apply to the entire config, like "the config of all the robot's resources". I almost think we should just delete this term. If not, it could still apply to a remote config, a module config, etc., so maybe broaden it.
```suggestion
The configuration element of a {{< glossary_tooltip term_id="resource" text="resource" >}} such as a {{< glossary_tooltip term_id="component" text="component" >}} or a {{< glossary_tooltip term_id="service" text="service" >}}.
```

File: docs/appendix/glossary/resource-config.md [hunk 12]
```diff
@@ -2,8 +2,10 @@
 title: Resource Config
 id: resource-config
 full_link:
-short_description: The configuration element of either a component or a service. Typically expressed in JSON.
+short_description: The configuration element of either a component or a service.
```
Comment (JessamyT) [inline docs/appendix/glossary/resource-config.md:5]:
```suggestion
short_description: The configuration element of a component, service or other resource.
```
I'm actually not 100% sure whether we want "resource" to include modules and remotes, or if those indirectly fall within components and services anyway (since modules are components or services, and remotes provide access to components and services of other robots). So this suggestion and the one below should be ignored if we just want to keep it to components and services.

File: docs/appendix/glossary/resource.md [hunk 13]
```diff
@@ -2,10 +2,12 @@
...
-short_description: Resources are individual, addressable elements of a robot (RDK definition) operated by parts.
+short_description: Resources are individual, addressable elements of a robot such as components or services.
 aka:
 ---
 
-Resources are individual, addressable elements of a robot (RDK definition) operated by parts.
+Resources are individual, addressable elements of a robot.
 
-Parts operate two types of Resources: physical {{< glossary_tooltip term_id="component" text="components" >}} and software services. Each part has local resources, and also surfaces remote resources when a {{< glossary_tooltip term_id="remote" text="remote">}} is established to another part. The capabilities of each resource are exposed through the part’s API.
+{{< glossary_tooltip term_id="part" text="Parts" >}} operate two types of Resources: physical {{< glossary_tooltip term_id="component" text="components" >}} and {{< glossary_tooltip term_id="service" text="software services" >}}.
```
Comment (JessamyT) [inline docs/appendix/glossary/resource.md:11]:
```suggestion
{{< glossary_tooltip term_id="part" text="Parts" >}} operate two types of resources: physical {{< glossary_tooltip term_id="component" text="components" >}} and {{< glossary_tooltip term_id="service" text="software services" >}}.
```
Need to decide whether modules, remotes and/or processes count as resources (see comment above), but maybe leave for now idk

File: docs/appendix/glossary/robot-config.md [hunk 14]
```diff
@@ -2,8 +2,10 @@
 title: Robot Config
 id: robot-config
 full_link:
-short_description: The complete configuration of a single robot part. Typically expressed in JSON.
+short_description: The complete configuration of a single robot part.
 aka:
 ---
 
-The complete configuration of a single robot part. Typically expressed in JSON.
+The complete configuration of a single robot part.
```
Comment (JessamyT) [inline docs/appendix/glossary/robot-config.md:9]:
```suggestion
The complete configuration of a single robot {{< glossary_tooltip term_id="part" text="part" >}}.
```

File: docs/appendix/glossary/robot-config.md
Comment (JessamyT) [inline docs/appendix/glossary/robot-config.md:1]: People could get wildly confused between this and "resource-config," especially if they don't understand the difference between a "part" and a "component." So if we want to delete one, I'd advocate for keeping robot config and deleting resource config. We don't talk about resource config very often anyway.

File: docs/appendix/glossary/viam-robot-api.md [hunk 15]
```diff
@@ -2,14 +2,15 @@
...
 
-The de facto description of how all resources can be communicated with.
+The specification for communication with resources.
 
-- The RDKs/SDKs implement this server side.
-- The RDKs/SDKs use this to act as clients.
-- Currently expressed as a collection of Protocol Buffer files.
-- Does not mandate gRPC as the transport mechanism.
-  - However, all RDKs/SDKs written by Viam use gRPC.
+- The RDKs/SDKs implement the Viam Robot API server-side.
```
Comment (JessamyT) [inline docs/appendix/glossary/viam-robot-api.md:11]: Is it supposed to be capitalized? Seems like it wouldn't be

File: docs/appendix/glossary/viam-robot-api.md [hunk 16]
```diff
@@ -2,14 +2,15 @@
 title: Viam Robot API
 id: viam-robot-api
 full_link:
-short_description: The de facto description of how all resources can be communicated with.
+short_description: The specification for communication with resources.
 aka:
 ---
 
-The de facto description of how all resources can be communicated with.
+The specification for communication with resources.
```
Comment (JessamyT) [inline docs/appendix/glossary/viam-robot-api.md:9]: Should it be "interface" instead of "description"? Not sure

File: docs/components/_index.md [hunk 17]
```diff
@@ -7,6 +7,7 @@ type: docs
 description: "In-depth information on the various components available within the Viam system."
 ---
 
-The term *component* describes a piece of hardware that a computer controls.
+The term *component* describes a resource that represents a physical component in a robot which a computer controls.
```
Comment (JessamyT) [inline docs/components/_index.md:10]: This is hard to parse and uses "component" to define itself
```suggestion
A *component* represents a physical piece of electrical or electromechanical hardware in a robot.
```

File: docs/components/_index.md [hunk 18]
```diff
@@ -7,6 +7,7 @@ type: docs
...
+
 The following are the component types that Viam natively supports.
```
Comment (JessamyT) [inline docs/components/_index.md:12]:
```suggestion
See the component types that Viam natively supports [here](/components/).
```

File: docs/manage/fleet/_index.md [hunk 19]
```diff
@@ -47,6 +47,16 @@ Role Based Access Control (RBAC) ia a way to enforce security in the [Viam app](
 
 A user can have one or more roles, granting the user the respective permissions of each role.
 
+## Robots
```
Comment (JessamyT) [inline docs/manage/fleet/_index.md:50]: Seems like this should be pared down since it's exactly the same as what is in the /robots page.

Also why is this below permissions? It seems like it should be above locations since in locations it says every robot belongs to a location

**Review by JessamyT — CHANGES_REQUESTED** [review]

File: docs/appendix/glossary/board.md (hunk 5 above)
Comment (npentrel) [reply docs/appendix/glossary/board.md:12]: Happy to remove

Comment (npentrel) [inline docs/appendix/glossary/board.md:12]:
```suggestion
```

File: docs/appendix/glossary/rdk.md (hunk 9 above)
Comment (npentrel) [reply docs/appendix/glossary/rdk.md:5]: Formatting in these doesn't do anything I'm afraid

File: docs/appendix/glossary/viam-robot-api.md (hunk 15 above)
Comment (npentrel) [reply docs/appendix/glossary/viam-robot-api.md:11]: I think so. It seems to me a big enough concept to deserve capitalization. We also only seems to use this term in the SDK page

File: docs/manage/fleet/_index.md (hunk 19 above)
Comment (npentrel) [reply docs/manage/fleet/_index.md:50]: going to do that with the fleet reorg PR I think - that moves everything even more

File: docs/appendix/glossary/robot-config.md
Comment (npentrel) [reply docs/appendix/glossary/robot-config.md:1]: ok

File: docs/appendix/glossary/resource.md (hunk 13 above)
Comment (npentrel) [reply docs/appendix/glossary/resource.md:11]: yeah let's leave for now if we're not sure

File: docs/appendix/glossary/viam-robot-api.md (hunk 16 above)
Comment (JessamyT) [reply docs/appendix/glossary/viam-robot-api.md:9]: Mm yes sorry I meant description

**Review by JessamyT — APPROVED** [review]
LGTM % maybe move the "Robots" section of fleet/_index to the beginning of the doc instead of so far down
